File: Shantanu_Mamgain/Prefix-Sum-Leetcode/Find-Pivot-Index.java
class Solution {
    public int pivotIndex(int[] nums) { 
        for(int i = 1; i<nums.length; i++){
            nums[i] += nums[i-1];
        }

        for(int i =0; i<nums.length;i++){
            if(i==0){
                if(nums[i]==nums[nums.length-1]){
                    return i;
                }
            }
            else{
                if(nums[i]==nums[nums.length-1]-nums[i-1]){
                    return i;
                }
            }
        }

        return -1;
    }
}

File: AdityaKumar1010707/PrefixSum & SuffixSum/findPivotIndex.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int sum = 0;
        for (int it : nums) {
            sum += it;
        }

        int size = nums.size();
        int l_sum = 0;
        int r_sum = sum;
        int bal_index = -1;

        for (int i = 0; i < size; i++) {
            r_sum = sum - l_sum - nums[i];
            if (l_sum == r_sum) {
                bal_index = i;
                break;
            } else {
                l_sum += nums[i];
            }
        }

        return bal_index;
    }
};

int main() {
    int n;
    cout << "Enter number of elements: ";
    cin >> n;

    vector<int> nums(n);
    cout << "Enter elements: ";
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }

    Solution obj;
    int result = obj.pivotIndex(nums);

    cout << "Pivot Index: " << result << endl;

    return 0;
}
